### ISSUE TITLE
mini.files closes and does not reopen

### ISSUE THREAD

**Issue by CMOISDEAD** [AstroNvim/AstroNvim#2688]
### Checklist

- [X] I have searched through the AstroNvim docs
- [X] I have searched through the existing issues of AstroNvim
- [X] I have searched the existing issues of plugins related to this issue
- [X] I have run `:Lazy update` and are on the latest version of AstroNvim
- [X] I can replicate the bug with the minimal `repro.lua` provided below

### Operating system/version

Arch Linux

### Terminal/GUI

Alacritty

### AstroNvim Health

AstroNvim ~
- AstroNvim Version: v4.23.8
- Neovim Version: v0.10.1
- OK Using stable Neovim >= 0.9.5
- OK `git` is installed: Used for core functionality such as updater and plugin management
- OK `xdg-open` is installed: Used for `gx` mapping for opening files with system opener (Optional)
- OK `rg` is installed: Used for Telescope `live_grep` picker, `<Leader>fw` and `<Leader>fW` by default (Optional)
- OK `lazygit` is installed: Used for mappings to pull up git TUI (Optional)
- OK `node` is installed: Used for mappings to pull up node REPL (Optional)
- WARNING `gdu/gdu_windows_amd64.exe` is not installed: Used for mappings to pull up disk usage analyzer (Optional)
- WARNING `btm` is not installed: Used for mappings to pull up system monitor (Optional)
- OK `python` is installed: Used for mappings to pull up python REPL (Optional)

### Describe the bug

When I close mini.files with the key q it closes and does not reopen even if I run the lua command. It sometimes happens when it just opens and closes, but the way it happens most of the time is when you open mini.files for the first time and close it instantly, then it doesn't open anymore. Another way is to open mini.files and move to a folder and use the `<` key to trim and close. Sometimes also when you repeat the same process but with backspace.

### Steps to Reproduce

1. install mini.file
2. toggle with if not require("mini.files").close() then require("mini.files").open() end
3. Open with the key previously bound and close with q.
4. mini.files just doesn't open again.

### Expected behavior

mini.files works as expected and no matter how it is closed it will come back and open.

### Additional Context

I use the astrocomunnity version and it also has this problem, I decided to install the plugin on a neovim without plugins and without astronvim and it worked as expected however when installing it on a newly installed astronvim the problem occurred so I think it must be a problem with astronvim.

### Repro

```Lua
return {
  "echasnovski/mini.files",
  dependencies = {
    "nvim-tree/nvim-web-devicons",
    {
      "AstroNvim/astrocore",
      opts = {
        mappings = {
          n = {
            -- ["<tab>"] no matter the keybind
            ["<Leader>e"] = {
              function()
                if not require("mini.files").close() then require("mini.files").open() end
              end,
              desc = "Explorer",
            },
          },
        },
      },
    },
  },
  specs = {
    { "neo-tree.nvim", optional = true, enabled = false },
  },
  opts = {},
}
```

**Comment by mehalter** [comment]
Please next time use the minimal `repro.lua` template as provided so that the repro provided is minimal.

Here is what you meant:

```lua
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  -- stylua: ignore
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", "--branch=stable", lazypath })
end
vim.opt.rtp:prepend(vim.env.LAZY or lazypath)

-- install plugins
local plugins = {
  { "AstroNvim/AstroNvim", import = "astronvim.plugins" },
  {
    "echasnovski/mini.files",
    dependencies = {
      "nvim-tree/nvim-web-devicons",
      {
        "AstroNvim/astrocore",
        opts = {
          mappings = {
            n = {
              -- ["<tab>"] no matter the keybind
              ["<Leader>e"] = {
                function()
                  if not require("mini.files").close() then
                    require("mini.files").open()
                  end
                end,
                desc = "Explorer",
              },
            },
          },
        },
      },
    },
    specs = {
      { "neo-tree.nvim", optional = true, enabled = false },
    },
    opts = {},
  },
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- add anything else here (autocommands, vim.filetype, etc.)
```

**Comment by mehalter** [comment]
Using the above minimal `repro.lua` this bug is not replicatable given the steps provided. `mini.files` is able to be closed and opened no problem

**Comment by mehalter** [comment]
Here is a full recording of a completely fresh install with the above: https://asciinema.org/a/re4K7sG0PwOf5ZrMqG7wVAwUy

**Comment by mehalter** [comment]
Also tested with AstroCommunity using the following:

```lua
-- save as repro.lua
-- run with nvim -u repro.lua
-- DO NOT change the paths
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "runtime", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  -- stylua: ignore
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", "--branch=stable", lazypath })
end
vim.opt.rtp:prepend(vim.env.LAZY or lazypath)

-- install plugins
local plugins = {
  { "AstroNvim/AstroNvim", import = "astronvim.plugins" },
  "AstroNvim/astrocommunity",
  { import = "astrocommunity.file-explorer.mini-files" },
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- add anything else here (autocommands, vim.filetype, etc.)
```

and it also works as intended. Here is another recording similar to the one above: https://asciinema.org/a/A62grjIW2CMU619NYqpQdrXG8

**Comment by CMOISDEAD** [comment]
Hi thanks for the quick reply and sorry for not using the template 😅, I will keep it in mind next time. Look run `nvim -u repro.lua` with the content you gave me and I did experience the bug, try going into multiple nested directories and use the `<` key. Do it a few times. in this video it took a while but it happened in the end I could no longer open `mini.files`. similarly it happens when closing `mini.files` with `q` but it happens sometimes.

Sorry for my english.

https://asciinema.org/a/2myGSxxS3suPtazhdnETcVBYo

**Comment by mehalter** [comment]
Thanks for the recording and explaining! While I cannot replicate this, it looks like it is a bug in `mini.files` since we would be doing nothing to affect it's internal state and whether or not `require("mini.files").close()` returns true or false. Sorry I'm not much help! I would recommend seeing if you can replicate the bug from no configuration and just `mini.files` added and then working backwards from there! If you are able to figure out that this is indeed due to an AstroNvim bug definitely report back and I can re-open this. In the mean time I really don't think there's anything I can do here. Sorry!

**Comment by CMOISDEAD** [comment]
Thank you for your input and suggestions! I’ll definitely take a closer look and try to replicate the issue as you recommended. I really appreciate your help, and if I discover that it’s related to AstroNvim, I’ll be sure to report back. Thanks again!

**Comment by mehalter** [comment]
@CMOISDEAD just to help out, I can replicate the exact same issue without AstroNvim using the following minimal `repro.lua` with just `nvim -u repro.lua`

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

vim.g.mapleader = " "

require("lazy.minit").repro({
  spec = {
    {
      "echasnovski/mini.files",
      keys = {
        {
          "<Leader>e",
          function()
            if not require("mini.files").close() then
              require("mini.files").open()
            end
          end,
        },
      },
      dependencies = { "nvim-tree/nvim-web-devicons" },
      opts = {},
    },
  },
})
```

**Comment by mehalter** [comment]
I started looking into trying to make a fix PR to `mini.files` and while I don't have a real fix together, I did find that this is because `mini.files` does not play well with `:q` or `:close`. It basically hard requires that you use it's own `require("mini.files").close()` which is pretty unfortunate.

This does sorta expose a potential issue where our autocommand for setting up `q` to close any non-file buffer can potentially overwrite existing buffer local `q` mappings. This is probably not ideal behavior because if something does map `q` locally to the buffer we should trust it was intentional. I just released v4.23.10 which should resolve this (commit https://github.com/AstroNvim/AstroNvim/commit/a368bd203220392589ac60deb6d47d02db9c7f5c).

This should ideally fix the bad `mini.files` behavior, but doesn't resolve the underlying issue in the plugin
